File: app/src/main/java/uk/co/sundroid/LocaterListener.java
package uk.co.sundroid;

import uk.co.sundroid.domain.LocationDetails;
import android.os.Looper;

public interface LocaterListener {
	
	void locationError();
	
	void locationTimeout();
	
	void locationReceived(LocationDetails locationDetails);
	
	Object getSystemService(String id);
	
	Looper getMainLooper();

}

File: app/src/main/kotlin/uk/co/sundroid/activity/LocaterListener.kt
package uk.co.sundroid.activity

import uk.co.sundroid.domain.LocationDetails
import android.os.Looper

interface LocaterListener {

    fun getMainLooper(): Looper?

    fun locationError(status: LocaterStatus)

    fun locationReceived(locationDetails: LocationDetails)

    fun getSystemService(id: String): Any?

}
